drivers/mspi/mspi_ambiq_ap4.c

```diff
@@ -0,0 +1,1530 @@
+/*
+ * Copyright (c) 2024, Ambiq Micro Inc. <www.ambiq.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ambiq_mspi_controller
+
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_instance.h>
+LOG_MODULE_REGISTER(mspi_ambiq_ap4);
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/mspi.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/irq.h>
+
+#include "mspi_ambiq.h"
+
+#define DT_SIZE_K(x) ((x) * 1024)
+#define DT_SIZE_M(x) ((x) * 1024 * 1024)
```

Comment:
This is in zephyr\dts\common\mem.h

Reply:
Yes, that was easy to find, however, getting that header to compile in anything but a dtsi file is not working.